### ISSUE TITLE
Android 4.2: "libSDL2_image.so" needed by "libmain.so" not found

### ISSUE THREAD

**Issue by byaka** [zielmicha/freeciv-android#20]
Cant find solution for this... Any ideas?

**Comment by javaxubuntu** [comment]
I've tried to investigate on this with adb. libSDL2_image.so is in the apk, and my phone don't have this problem.
To run adb, install [Freeciv-2.6.0-debug.apk](https://github.com/zielmicha/freeciv-android/releases/download/v2.6.0/Freeciv-2.6.0-debug.apk)
If I don't start the app, the **.so** files are already extracted:
Connect your phone with your computer, enable usb debugging and run: adb shell
```
shell@C5503:/ $ run-as pl.org.zielinscy.freeciv.debug
shell@C5503:/data/data/pl.org.zielinscy.freeciv.debug $ ls -l
lrwxrwxrwx install  install           2018-08-02 10:07 lib -> /data/app/pl.org.zielinscy.freeciv.debug-1/lib/arm
shell@C5503:/data/data/pl.org.zielinscy.freeciv.debug $ ls -l /data/app/pl.org.zielinscy.freeciv.debug-1/lib
drwxr-xr-x system   system            2018-08-02 10:07 arm
shell@C5503:/data/data/pl.org.zielinscy.freeciv.debug $ cd lib
shell@C5503:/data/data/pl.org.zielinscy.freeciv.debug/lib $ ls -l
-rwxr-xr-x system   system     888480 1979-12-31 09:44 libSDL2.so
-rwxr-xr-x system   system     427508 1979-12-31 09:44 libSDL2_image.so
-rwxr-xr-x system   system     282520 1979-12-31 09:44 libSDL2_mixer.so
-rwxr-xr-x system   system     382452 1979-12-31 09:44 libSDL2_ttf.so
-rwxr-xr-x system   system    1893360 1979-12-31 09:44 libfreeciv-client.so
-rwxr-xr-x system   system     276216 1979-12-31 09:44 libmain.so
-rwxr-xr-x system   system    1470024 1979-12-31 09:44 libpython2.7.so
shell@C5503:/data/data/pl.org.zielinscy.freeciv.debug/lib $ echo >test.so
sh: can't create test.so: Permission denied
1|shell@C5503:/data/data/pl.org.zielinscy.freeciv.debug/lib $ mkdir test.so
mkdir failed for test.so, Permission denied
255|shell@C5503:/data/data/pl.org.zielinscy.freeciv.debug/lib $ echo >>libSDL2_image.so
sh: can't create libSDL2_image.so: Permission denied
1|shell@C5503:/data/data/pl.org.zielinscy.freeciv.debug/lib $ touch libSDL2_image.so
255|shell@C5503:/data/data/pl.org.zielinscy.freeciv.debug/lib $ 
```
I conclude that these files are automatically extracted by Android, because the app don't have any write permission in the lib directory.
I have no idea what happens when you have this error. You can also try the old apk [here](https://github.com/zielmicha/freeciv-android/issues/16) because I don't know if this bug is a regression.

**Comment by javaxubuntu** [comment]
Also, in order to know if this bug is device-dependant, I need to know the exact model number and android version, for example in my phone:
About phone => Model number: C5503
Android version: 5.1.1 (vendor version)

**Comment by javaxubuntu** [comment]
Freeciv needs at least Android 4.2 to run. By mistake it was installable on older devices but is it not compatible with older Android versions.
If this bug happens on newer versions, please reopen this issue.

**Comment by javaxubuntu** [comment]
For an unknown reason this issue also affects Android 4.2, although it should be compatible with Android 4.2.

**Comment by byaka** [comment]
> For an unknown reason this issue also affects Android 4.2, although it should be compatible with Android 4.2.

Yep. You still need debug-info from second post?

**Comment by javaxubuntu** [comment]
I don't need these debug-info because it wasn't the correct way to understand the problem. Now I've found a similar issue [here](https://github.com/kivy/python-for-android/issues/645) and its solution. [The old freeciv app did load the libraries in the right order](https://github.com/zielmicha/freeciv-android/blob/1.1.1/android/project/src/org/libsdl/app/SDLActivity.java). I'll build a new apk in a few days, I home it'll fix it.

**Comment by javaxubuntu** [comment]
I've fixed the bug and I've created a new apk [here](https://github.com/zielmicha/freeciv-android/releases/tag/v2.6.0.6). This apk will come to Google Play soon. Note that Freeciv-2.6.0.6-autosigned.apk I've created is signed with my own key, not zielmicha's key. Therefore if you want to test Freeciv-2.6.0.6-autosigned.apk before it's on Google Play, you'll have to uninstall the Freeciv version you got from Google Play. Or you can wait that zielmicha sign the apk with his key and push it to Google Play.

**Comment by javaxubuntu** [comment]
You can now install the beta version [from Google Play Store](https://play.google.com/apps/testing/pl.org.zielinscy.freeciv). Can you confirm if this bug is fixed?
Best regards.

**Comment by byaka** [comment]
Problem fixed, thx!

**Comment by javaxubuntu** [comment]
I would like to rectify my previous comments above: since [this release](https://github.com/zielmicha/freeciv-android/releases/tag/v2.6.0.6_pre-kitkat), this bug is now fixed for all android versions between 2.3 and 4.3, and this release has been pushed to Google Play Store, so this bug should not happen anymore on any device. If it still happens, please reopen this issue.
